### PR TITLE
burners-balance. now player drop burners when left game

### PR DESCRIPTION
**Pull request by PlayerNoon**
### Brief description of the changes:
Player will spill all his burners on surface on_player_left_game

### Tested Changes:
- [x] I've tested the changes locally or with people.
- [ ] I've not tested the changes.

### CODE REVIEW

**Review by clifffrey — CHANGES_REQUESTED** [review]
I don't understand why we do it when the player leaves, but not if they just click the "spectate" button. Should this happen on spectate as well?

I'm also not really sure if this change is actually worthwhile... the behavior seems confusing/surprising to me.  I'd really prefer that we just check for the case where the player has exactly 10 burners, and remove the burners (and remove the player from `global.got_burners` as well).

**Comment by PlayerNoon** [comment]
> I don't understand why we do it when the player leaves, but not if they just click the "spectate" button. Should this happen on spectate as well?
> 
> I'm also not really sure if this change is actually worthwhile... the behavior seems confusing/surprising to me. I'd really prefer that we just check for the case where the player has exactly 10 burners, and remove the burners (and remove the player from `global.got_burners` as well).

1. Totally forgot about spectate, i will add this.
2. if i am adding burners always for 2 players then if i want remove then i have to do this for both players also. i can't do that in case when 1 of them have already placed them. we end in situation where team with 2 players have 20 burners and team with 1 player have 10 burners. 
In case of dropping them on the surface the team with 1 player will have 20 burners and team with 2 players also 20 burners.

File: maps/biter_battles_v2/gui.lua [hunk 1]
```diff
@@ -374,6 +374,26 @@ local get_player_data = function (player, remove)
...
+local function drop_burners(player,forced_join)
+	if forced_join then 
+		global.got_burners[player.name] = nil
+		return
+	end
+	if global.training_mode or not (global.bb_settings.burners_balance) then 		
+		return
+	end			
+	local inventory = player.get_item_count("burner-mining-drill")	
+	if not (inventory == 0) then
```
Comment (clifffrey) [inline maps/biter_battles_v2/gui.lua:386]: you should write this as `if inventory ~= 0 then` or `if inventory > 0 then`

File: maps/biter_battles_v2/gui.lua [hunk 2]
```diff
@@ -374,6 +374,26 @@ local get_player_data = function (player, remove)
...
 
+local function drop_burners(player,forced_join)
+	if forced_join then 
+		global.got_burners[player.name] = nil
+		return
+	end
+	if global.training_mode or not (global.bb_settings.burners_balance) then 		
+		return
+	end			
+	local inventory = player.get_item_count("burner-mining-drill")
```
Comment (clifffrey) [inline maps/biter_battles_v2/gui.lua:385]: rename `inventory` to `num_burners` or something. Otherwise I'd assume that this is `player.get_inventory()`

**Review by clifffrey — CHANGES_REQUESTED** [review]
This is just about ready. Thank you.

It would be great if the change consistently used ", " rather than just "," when separating arguments to functions, which is what most of the existing code does.

File: maps/biter_battles_v2/gui.lua (hunk 1 above)
Comment (PlayerNoon) [reply maps/biter_battles_v2/gui.lua:386]: yup

File: maps/biter_battles_v2/gui.lua (hunk 2 above)
Comment (PlayerNoon) [reply maps/biter_battles_v2/gui.lua:385]: right

**Review by clifffrey — APPROVED** [review]

**Pull request merged**